pyomo/core/plugins/transform/discrete_vars.py

```diff
@@ -114,7 +114,7 @@ def _apply_to(self, model, **kwds):
         _base_model_vars = model.component_data_objects(
             Var, active=True, descend_into=True )
         for var in _base_model_vars:
-            if var.domain in _discrete_relaxation_map and not var.is_fixed():
+            if any(type(var.domain) is type(domain) for domain in _discrete_relaxation_map) and not var.is_fixed():
                 fixed_vars.append(var)
```

Comment:
I think I would prefer:
```python
for var in _base_model_vars:
    if var.is_fixed():
        continue
    if var.domain not in _discrete_relaxation_map:
        # The following extra check is needed for cloned models until the management 
        # of Set singletons is resolved by #326.
        if all(type(var.domain) is not type(domain) for domain in _discrete_relaxation_map):
            continue
    fixed_vars.append(var)
    var.fix()
```

Reply:
Since we don't actually use the `_discrete_relaxation_map.values()` (we just compare against the keys), we could also consider the more simple
```python
for var in _base_model_vars:
    # Instead of checking against `_discrete_relaxation_map.keys()` we just check the item properties to fix #995 
    # When #326 has been resolved, we can check against the dict-keys again
    if not var.is_continuous() and not var.is_fixed():
        fixed_vars.append(var)
        var.fix()
```

Reply:
One thing to note of course is that neither of the proposed solutions fix the `TransformationFactory('core.relax_discrete')` defined in the same file, because that implementation actually needs to look up values from `_discrete_relaxation_map`.
It would of course be nice to fix that too, but without #326 we will probably need a fix that's not super pretty.

Reply:
I think moving to `is_continuous()` is a nice fix.

As for #326, it is closer than you might think: the bulk of the work is done, and I am just cleaning up set operations and finally getting to the Set initialization.  That said, the PR / PEP review might take a while.  I am hoping that it might make it on master by mid June, but we will see.

Reply:
Fixed in the `is_continuous()` in 72dbabaf64dfd7e216f8abd25aafe14b970d9c6c

Reply:
When looking at the tests, I noticed another thing:
The tests only check for `TransformationFactory('core.relax_integrality')` which just transforms all Variables to domain `Reals` (and sets the bounds) [source](https://github.com/Pyomo/pyomo/blob/master/pyomo/core/plugins/transform/relax_integrality.py) with no regard for the original domain. Note that this works also with cloned models, because no type lookup happens.
The method discussed earlier was `TransformationFactory('core.relax_discrete')`. This seems like redundant functionality.

Reply:
Assigned issue #998